widgetastic_manageiq/__init__.py

```diff
@@ -61,6 +61,58 @@
 from cfme.exceptions import ItemNotFound
 
 
+class SnapshotMemorySwitch(Widget):
+    """ This replaces the old-style checkbox when working with snapshot memory.
+
+    At first glance on the page it looks like a BootstrapSwitch but it really isn't.
+    Basic switch controls are implemented here along with fill and read methods,
+    so this should be usable as any other widget.
+    """
+
+    LOCATOR = u'//div[contains(@class, "bootstrap-switch-snap_memory")]'
+    ON_LOCATOR = u'//div[contains(@class, "bootstrap-switch-on")]'
+
+    def __init__(self, parent, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+
+    def __locator__(self):
+        return self.LOCATOR
+
+    def switch(self):
+        self.browser.click(self.LOCATOR)
+
+    @property
+    def is_set(self):
+        if self.browser.selenium.find_elements_by_xpath(self.ON_LOCATOR):
```

Comment:
Additionally, since the element isn't actually used, and you aren't catching exceptions when the element isn't present, This can just be `return self.browser.elements(self.ON_LOCATOR)`